src/Text/Markdown/SlamDown/Halogen/Component.purs

```diff
@@ -338,85 +380,125 @@ renderSlamDown config (SDS.SlamDownState state) =
 renderRadioButton label value checked = do
-  id ← Fresh.fresh
+  ident ← Fresh.fresh
   let renderedValue = SD.renderValue value
   pure $ HH.li_
     [ HH.input
         [ HP.checked checked
         , HP.type_ HP.InputRadio
-        , HP.id_ id
+        , HP.id_ ident
         , HP.name label
         , HP.value renderedValue
         , HE.onValueChange (HE.input_ (SDQ.ChangeRadioButton label value))
         ]
-    , HH.label [ HP.for id ] [ HH.text renderedValue ]
+    , HH.label [ HP.for ident ] [ HH.text renderedValue ]
     ]
 
 renderCheckBox
-  ∷ ∀ p v
+  ∷ ∀ m v
   . (SD.Value v)
   ⇒ String  -- label
   → v       -- value
-  → FreshRenderer p v Boolean
+  → FreshRenderer v m Boolean
 renderCheckBox label value checked = do
-  id ← Fresh.fresh
+  ident ← Fresh.fresh
   let renderedValue = SD.renderValue value
   pure $ HH.li_
     [ HH.input
         [ HP.checked checked
         , HP.type_ HP.InputCheckbox
-        , HP.id_ id
+        , HP.id_ ident
         , HP.name label
         , HP.value renderedValue
         , HE.onChecked (HE.input (SDQ.ChangeCheckBox label value))
         ]
-    , HH.label [ HP.for id ] [ HH.text renderedValue ]
+    , HH.label [ HP.for ident ] [ HH.text renderedValue ]
     ]
 
 renderDropDown
-  ∷ ∀ p v
+  ∷ ∀ m v
   . (SD.Value v)
   ⇒ String    -- id
   → String    -- label
   → L.List v  -- choices
   → M.Maybe v -- value
-  → H.HTML p (SDQ.SlamDownQuery v)
-renderDropDown id label ls sel =
+  → HTML v m
+renderDropDown ident label ls sel =
   HH.select
-    [ HP.id_ id
+    [ HP.id_ ident
     , HP.name label
     , HE.onSelectedIndexChange (HE.input (SDQ.ChangeDropDown label <<< L.index ls))
     ]
     $ A.fromFoldable $ M.maybe option option' sel <$> ls
   where
-    option ∷ v → H.HTML p (SDQ.SlamDownQuery v)
+    option ∷ v → HTML v m
     option value =
       let renderedValue = SD.renderValue value
       in HH.option [ HP.value renderedValue ] [ HH.text renderedValue ]
 
-    option' ∷ v → v → H.HTML p (SDQ.SlamDownQuery v)
+    option' ∷ v → v → HTML v m
     option' sel' value =
       let renderedValue = SD.renderValue value
       in HH.option [ HP.selected (value == sel'), HP.value renderedValue ] [ HH.text renderedValue ]
 
+data PickerType
+  = DatePicker
+  | TimePicker SD.TimePrecision
+  | DateTimePicker SD.TimePrecision
+
+textBoxToPicker ∷ ∀ f
+  . BrowserFeatures
+  → SD.TextBox f
+  → Maybe PickerType
+textBoxToPicker features tb = do
+  guard (features.inputTypeSupported $ SDIT.textBoxToInputType tb)
+  case tb of
+    SD.Date _ → Just DatePicker
+    SD.Time p _ → Just $ TimePicker p
+    SD.DateTime p _ → Just $ DateTimePicker p
+    _ → Nothing
+
+dateFormatString :: String
+dateFormatString = "YYYY-MMMM-DD"
+
+timeFormatString :: SD.TimePrecision → String
+timeFormatString SD.Minutes = "HH:mm"
+timeFormatString SD.Seconds = "HH:mm:ss"
+
+datePickerFormat :: DatePickerF.Format
+datePickerFormat = unsafePartial fromRight
+  $ DatePickerF.fromString dateFormatString
```

Comment:
Something I've been thinking of for a while... why do we use these string formats anywhere in PS code for the formatter/printer formats? It seems crazy when doing `Hours24 : MinutesTwoDigits : SecondsTwoDigits : Nil` is typed, requires no `unsafePartial fromRight`, and doesn't involve going through a parser etc. on startup. 😆 

I'll open an issue on `formatters` pointing this out actually, we should de-emphasise the use of the string formats in the docs in favour of using the types properly instead.